troi.py

```diff
@@ -18,30 +17,39 @@ def cli():
 @click.argument('args', nargs=-1)
 @click.option('--debug', '-d', is_flag=True, default=False)
 def playlist(patch, args, debug):
-
+    """Generate a playlist using a patch"""
     if debug:
-        print("- debug mode on")
+        print("- debug mode on", file=sys.stderr)
 
     patches = troi.utils.discover_patches("patches")
-    if not patch in patches:
-        print("Cannot load patch '%s'. Use the list command to get a list of available patches." % patch)
-        quit()
+    if patch not in patches:
+        print("Cannot load patch '%s'. Use the list command to get a list of available patches." % patch,
+              file=sys.stderr)
+        sys.exit(1)
 
     patch = patches[patch]()
     inputs = patch.inputs()
 
+    required_inputs = [input for input in inputs if not input['optional']]
+    num_required_inputs = len(required_inputs)
+    if num_required_inputs > len(args):
```

Comment:
I see that you removed this in c6df376ba6, so I assume that there's still a bug that I ran into. I was finding that if I run a patch that has 2 required arguments and I don't specify any, `args` below is empty, and so `args[i]` fails. Will look into it again in more detail before merging this